<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GestureMusicPlayer</class>
 <widget class="QMainWindow" name="GestureMusicPlayer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1016</width>
    <height>953</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>950</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Gesture Music Player</string>
  </property>
  <property name="windowIcon">
   <iconset resource="utils/music_res.qrc">
    <normaloff>:/img/images/CD.png</normaloff>:/img/images/CD.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_15">
       <item>
        <widget class="QFrame" name="gesture_frame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="camera_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>500</width>
              <height>300</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>19</pointsize>
             </font>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true">QLabel {
    background-color: #4f4f4f;  /* 背景顏色為黑色 */
    color: white;  /* 文字顏色為白色 */
    border-radius: 12px;  /* 添加圓角效果，10px 可根據需要調整 */
    text-align: center;  /* 確保文字居中 */
    qproperty-alignment: 'AlignCenter';  /* 確保 QLabel 內容完全置中 */
	padding: 10px
}</string>
            </property>
            <property name="text">
             <string>Lens screen not found...</string>
            </property>
            <property name="textFormat">
             <enum>Qt::AutoText</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="gesture_instruction_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <weight>75</weight>
              <bold>true</bold>
              <kerning>true</kerning>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">line-height: 1.0;
border: 2px solid black;      
 border-radius: 10px;         </string>
            </property>
            <property name="lineWidth">
             <number>-1</number>
            </property>
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:14pt; font-weight:400;&quot;&gt;~~~Gesture Instructions~~~&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:400;&quot;&gt;ok: Start Music&lt;br/&gt;0: Pause/Unpause Music&lt;br/&gt;Pinky: Stop Music&lt;br/&gt;1~9: ⏩ n × 10s&lt;br/&gt;lotus+0~9: ⏪ n × 10s&lt;br/&gt;Thumb Up/Down: Volume Up/Down&lt;br/&gt;Thumb Left/Rignt: Previous/Next Song&lt;br/&gt;Rock!: 100% Volume&lt;br/&gt;▼ Latest Gesture Recognition Results &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="textFormat">
             <enum>Qt::RichText</enum>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="info_textbox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>50</height>
             </size>
            </property>
            <property name="baseSize">
             <size>
              <width>0</width>
              <height>51</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>20</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">QTextEdit {
    background-color: #f0f0f0;
    color: #000000;
    border: 2px solid #cccccc;
    border-radius: 10px;
    padding: 8px;
}</string>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Sunken</enum>
            </property>
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarAsNeeded</enum>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Arial'; font-size:20pt; font-weight:600; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; line-height:120%;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="current_song_label">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>-1</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">    QLabel {
        background-color: grey;       /* 灰色背景 */
        color: yellow;                /* 黄色文字 */
        font-size: 16px;              /* 可选：设置字体大小 */
        border: 2px solid black;      /* 边框样式 */
        border-radius: 10px;          /* 圆角半径 */
        padding: 5px;                 /* 可选：增加文字的内边距 */
    }</string>
            </property>
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;Playing Nothing&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="listWidget">
         <property name="font">
          <font>
           <pointsize>13</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="cursor" stdset="0">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="styleSheet">
          <string notr="true">QListView {
    outline: none;
    border: none;
    background-color: #4f4f4f; /* 整體背景淺灰色 */
    border-radius: 12px; /* 外框圓角 */
}

#listWidget::item {
    background-color: #4f4f4f; /* 項目背景色 */
    color: #ffffff; /* 文字顏色 */
    border: none;
    padding: 8px;
    margin: 4px; /* 增加間距以突顯圓角效果 */
    border-radius: 12px; /* 每個項目圓角 */
}

#listWidget::item:hover {
    background-color: #6c6c6c; /* 懸停時的較亮背景色 */
    border-radius: 12px; /* 保持圓角 */
}

#listWidget::item:selected {
    background-color: #009688; /* 選中背景色 */
    color: #ffffff; /* 選中後文字顏色 */
    border: none;
    border-radius: 12px; /* 選中效果圓角 */
}
</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="volume_music_ctrl_frame">
      <property name="styleSheet">
       <string notr="true">background-color: rgb(222, 222, 222);</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item alignment="Qt::AlignLeft">
        <widget class="QFrame" name="volume_frame">
         <property name="font">
          <font>
           <pointsize>1</pointsize>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="label">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Volume:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="volumeSlider">
            <property name="minimumSize">
             <size>
              <width>250</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>250</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">QSlider {
    border-radius: 10px;
}
QSlider::groove:horizontal {
    height: 5px;
    background: #fff; /* 白色 */
}
QSlider::handle:horizontal {
    background: #f00;
    width: 16px;
    height: 16px;
    margin: -6px 0;
    border-radius: 8px;
}
QSlider::sub-page:horizontal {
    background: qlineargradient(
        spread: pad, 
        x1: 0, y1: 0.5, x2: 1, y2: 0.5, 
        stop: 0 #0000ff,   /* 深藍 */
        stop: 1 #87cefa    /* 淺藍 */
    );
}
</string>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
            <property name="invertedControls">
             <bool>false</bool>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBothSides</enum>
            </property>
            <property name="tickInterval">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="volume_label">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>11</pointsize>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>50</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="music_ctrl_frame">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QFrame" name="ctrl_btn_frame">
            <property name="maximumSize">
             <size>
              <width>250</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QFrame" name="inner_ctrl_btn_frame">
               <property name="frameShape">
                <enum>QFrame::NoFrame</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <property name="spacing">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QPushButton" name="playpushButton">
                  <property name="cursor">
                   <cursorShape>PointingHandCursor</cursorShape>
                  </property>
                  <property name="toolTip">
                   <string>Play</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset resource="utils/music_res.qrc">
                    <normaloff>:/img/images/Play.png</normaloff>:/img/images/Play.png</iconset>
                  </property>
                  <property name="iconSize">
                   <size>
                    <width>24</width>
                    <height>24</height>
                   </size>
                  </property>
                  <property name="flat">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pausepushButton">
                  <property name="cursor">
                   <cursorShape>PointingHandCursor</cursorShape>
                  </property>
                  <property name="toolTip">
                   <string>Pause/Unpause</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset resource="utils/music_res.qrc">
                    <normaloff>:/img/images/Pause.png</normaloff>:/img/images/Pause.png</iconset>
                  </property>
                  <property name="iconSize">
                   <size>
                    <width>24</width>
                    <height>24</height>
                   </size>
                  </property>
                  <property name="flat">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="stoppushButton">
                  <property name="cursor">
                   <cursorShape>PointingHandCursor</cursorShape>
                  </property>
                  <property name="toolTip">
                   <string>Stop</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset resource="utils/music_res.qrc">
                    <normaloff>:/img/images/Stop.png</normaloff>:/img/images/Stop.png</iconset>
                  </property>
                  <property name="iconSize">
                   <size>
                    <width>24</width>
                    <height>24</height>
                   </size>
                  </property>
                  <property name="flat">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="previouspushButton">
                  <property name="cursor">
                   <cursorShape>PointingHandCursor</cursorShape>
                  </property>
                  <property name="toolTip">
                   <string>Previous</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset resource="utils/music_res.qrc">
                    <normaloff>:/img/images/Previous.png</normaloff>:/img/images/Previous.png</iconset>
                  </property>
                  <property name="iconSize">
                   <size>
                    <width>24</width>
                    <height>24</height>
                   </size>
                  </property>
                  <property name="flat">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="nextpushButton">
                  <property name="cursor">
                   <cursorShape>PointingHandCursor</cursorShape>
                  </property>
                  <property name="toolTip">
                   <string>Next</string>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset resource="utils/music_res.qrc">
                    <normaloff>:/img/images/Next.png</normaloff>:/img/images/Next.png</iconset>
                  </property>
                  <property name="iconSize">
                   <size>
                    <width>24</width>
                    <height>24</height>
                   </size>
                  </property>
                  <property name="flat">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="music_progress_frame">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="start_time_label">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>11</pointsize>
                </font>
               </property>
               <property name="text">
                <string>00:00:00</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="musicSlider">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">QSlider {
    border-radius: 10px;
}
QSlider::groove:horizontal {
    height: 5px;
    background: #fff; /* 白色 */
}
QSlider::handle:horizontal {
    background: #f00;
    width: 16px;
    height: 16px;
    margin: -6px 0;
    border-radius: 8px;
}
QSlider::sub-page:horizontal {
    background: qlineargradient(
        spread: pad, 
        x1: 0, y1: 0.5, x2: 1, y2: 0.5, 
        stop: 0 #0000ff,   /* 深藍 */
        stop: 1 #87cefa    /* 淺藍 */
    );
}
</string>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="end_time_label">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>11</pointsize>
                </font>
               </property>
               <property name="text">
                <string>00:00:00</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1016</width>
     <height>34</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="cursor">
    <cursorShape>ArrowCursor</cursorShape>
   </property>
   <property name="styleSheet">
    <string notr="true">QMenuBar {
    background-color: #f0f0f0;  /* 設置背景顏色為淺灰色 */
    color: #000000;  /* 設置文字顏色為黑色 */
}

QMenuBar::item {
    background-color: transparent;  /* 項目背景透明 */
    color: #000000;  /* 項目文字顏色 */
    padding: 8px;  /* 設置項目內邊距 */
}

QMenuBar::item:selected {
    background-color: #c8c8c8;  /* 選中項目的背景顏色 */
    color: #000000;  /* 選中項目的文字顏色 */
}
</string>
   </property>
   <widget class="QMenu" name="menuMenu">
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="actionAdd_Songs"/>
    <addaction name="separator"/>
    <addaction name="actionRemove_Selected"/>
    <addaction name="separator"/>
    <addaction name="actionRemove_All"/>
   </widget>
   <addaction name="menuMenu"/>
  </widget>
  <action name="actionAdd_Songs">
   <property name="icon">
    <iconset resource="utils/music_res.qrc">
     <normaloff>:/img/images/Create.png</normaloff>:/img/images/Create.png</iconset>
   </property>
   <property name="text">
    <string>Add Songs</string>
   </property>
   <property name="toolTip">
    <string>Add songs to music player</string>
   </property>
   <property name="shortcut">
    <string>Alt+A</string>
   </property>
  </action>
  <action name="actionRemove_Selected">
   <property name="icon">
    <iconset resource="utils/music_res.qrc">
     <normaloff>:/img/images/Remove.png</normaloff>:/img/images/Remove.png</iconset>
   </property>
   <property name="text">
    <string>Remove Selected</string>
   </property>
   <property name="toolTip">
    <string>Remove Selected Song</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionRemove_All">
   <property name="icon">
    <iconset resource="utils/music_res.qrc">
     <normaloff>:/img/images/Delete.png</normaloff>:/img/images/Delete.png</iconset>
   </property>
   <property name="text">
    <string>Remove All</string>
   </property>
   <property name="toolTip">
    <string>Remove All Songs</string>
   </property>
   <property name="shortcut">
    <string>Alt+Del</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="utils/music_res.qrc"/>
 </resources>
 <connections/>
</ui>
